pkg/api/v1alpha1/cluster.go

```diff
@@ -674,6 +682,14 @@ func validateMirrorConfig(clusterConfig *Cluster) error {
 	if clusterConfig.Spec.RegistryMirrorConfiguration.InsecureSkipVerify && clusterConfig.Spec.DatacenterRef.Kind != SnowDatacenterKind {
 		return errors.New("insecureSkipVerify is only supported for snow provider")
 	}
+
+	if clusterConfig.Spec.RegistryMirrorConfiguration.Authenticate {
+		username := os.Getenv("REGISTRY_USERNAME")
+		password := os.Getenv("REGISTRY_PASSWORD")
+		if username == "" || password == "" {
+			return fmt.Errorf("username or password not set, Provide REGISTRY_USERNAME and REGISTRY_PASSWORD to use authenticated registry mirror")
+		}
```

Comment:
you can just use this predefined function https://github.com/aws/eks-anywhere/blob/main/pkg/config/registry.go#L8